express-notes-api/server.ts

```diff
@@ -0,0 +1,182 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+type Note = {
+  id: number;
+  content: string;
+};
+
+type Notes = Record<number, Note>;
+
+type Data = {
+  nextId: number;
+  notes: Notes;
+};
+
+async function readNotes() {
+  const jsonData = await readFile('./data.json', 'utf8');
+  const data: Data = JSON.parse(jsonData);
+  return data;
```

Comment:
You can actually combine these into one statement and still get type checking in one of two ways. I'll give you my preferred way first:
```
async function readNotes(): Promise<Data> {
  const jsonData = await readFile('./data.json', 'utf8');
  return JSON.parse(jsonData);
}
```
That is by far the best way because your function declares its return type. You should always declare the return type on your functions!

This next way uses a type assertion:
```
async function readNotes() {
  const jsonData = await readFile('./data.json', 'utf8');
  return JSON.parse(jsonData) as Data;
}
```